core/math/math_funcs.h

```diff
@@ -51,149 +51,129 @@ class Math {
 
-		return ::atan2(p_y,p_x);
+	static _ALWAYS_INLINE_ double exp(double p_x) { return ::exp(p_x); }
+	static _ALWAYS_INLINE_ float exp(float p_x) { return ::expf(p_x); }
 
-	}
+	static _ALWAYS_INLINE_ bool is_nan(double p_val) { return (p_val!=p_val); }
+	static _ALWAYS_INLINE_ bool is_nan(float p_val) { return (p_val!=p_val); }
 
-	static _ALWAYS_INLINE_ double deg2rad(double p_y) {
-
-		return p_y*Math_PI/180.0;
-	}
-
-	static _ALWAYS_INLINE_ double rad2deg(double p_y) {
-
-		return p_y*180.0/Math_PI;
+	static _ALWAYS_INLINE_ bool is_inf(double p_val) {
+	#ifdef _MSC_VER
+		return !_finite(p_val);
+	#else
+		return isinf(p_val);
+	#endif
 	}
-
-
-	static _ALWAYS_INLINE_ double sqrt(double p_x) {
-
-		return ::sqrt(p_x);
+	
+	static _ALWAYS_INLINE_ bool is_inf(float p_val) {
+	#ifdef _MSC_VER
+		return !_finite(p_val);
+	#else
+		return isinf(p_val);
+	#endif
 	}
+	
+	static _FORCE_INLINE_ double abs(double g) { return absd(g); }
+	static _FORCE_INLINE_ float abs(float g) { return absf(g); }
+	static _FORCE_INLINE_ int abs(int g) { return g > 0 ? g : -g; }
 
-	static _ALWAYS_INLINE_ double fmod(double p_x,double p_y) {
-
-		return ::fmod(p_x,p_y);
-	}
+	static _ALWAYS_INLINE_ double fposmod(double p_x,double p_y) { return (p_x>=0) ? Math::fmod(p_x,p_y) : p_y-Math::fmod(-p_x,p_y); }
+	static _ALWAYS_INLINE_ float fposmod(float p_x,float p_y) { return (p_x>=0) ? Math::fmod(p_x,p_y) : p_y-Math::fmod(-p_x,p_y); }
 
-	static _ALWAYS_INLINE_ double fposmod(double p_x,double p_y) {
+	static _ALWAYS_INLINE_ double deg2rad(double p_y) { return p_y*Math_PI/180.0; }
+	static _ALWAYS_INLINE_ float deg2rad(float p_y) { return p_y*Math_PI/180.0; }
 
-		if (p_x>=0) {
+	static _ALWAYS_INLINE_ double rad2deg(double p_y) { return p_y*180.0/Math_PI; }
+	static _ALWAYS_INLINE_ float rad2deg(float p_y) { return p_y*180.0/Math_PI; }
 
-			return fmod(p_x,p_y);
+	static _ALWAYS_INLINE_ double lerp(double a, double b, double c) { return a+(b-a)*c; }
+	static _ALWAYS_INLINE_ float lerp(float a, float b, float c) { return a+(b-a)*c; }
 
-		} else {
+	static _ALWAYS_INLINE_ double linear2db(double p_linear) { return Math::log( p_linear ) * 8.6858896380650365530225783783321; }
+	static _ALWAYS_INLINE_ float linear2db(float p_linear) { return Math::log( p_linear ) * 8.6858896380650365530225783783321; }
 
-			return p_y-fmod(-p_x,p_y);
-		}
+	static _ALWAYS_INLINE_ double db2linear(double p_db) { return Math::exp( p_db * 0.11512925464970228420089957273422 ); }
+	static _ALWAYS_INLINE_ float db2linear(float p_db) { return Math::exp( p_db * 0.11512925464970228420089957273422 ); }
 
-	}
-	static _ALWAYS_INLINE_ double floor(double p_x) {
+	static _ALWAYS_INLINE_ double round(double p_val) { return (p_val>=0) ? Math::floor(p_val+0.5) : -Math::floor(-p_val+0.5); }
+	static _ALWAYS_INLINE_ float round(float p_val) { return (p_val>=0) ? Math::floor(p_val+0.5) : -Math::floor(-p_val+0.5); }
 
-		return ::floor(p_x);
-	}
+	static double ease(double p_x, double p_c);
```

Comment:
In general, and as a rule, functions that are not inlined simply leave as double. The performance impact of calling a function is greater than the float/double conversion